contracts/staking/GoodFundManager.sol

```diff
@@ -189,103 +221,191 @@ contract GoodFundManager is DAOContract {
-			iToken.balanceOf(reserveAddress).sub(currentBalance);
-		//uint256 effectiveInterest =
-		//	interest.mul(avgEffectiveStakedRatio).div(DECIMAL1e18);
-		//uint256 interestDonated = interest.sub(effectiveInterest);
-        // Mints gd while the interest amount is equal to the transferred amount
-        uint256 gdUBI = GoodReserveCDai(reserveAddress).mintUBI(
-            iToken,
-            interest
-        );
-        //_staking.updateGlobalGDYieldPerToken(gdInterest, interest);
-        // Transfers the minted tokens to the given staking contract
-        IGoodDollar token = IGoodDollar(address(avatar.nativeToken()));
-        //if(gdInterest > 0)
-        //    require(token.transfer(address(_staking), gdInterest),"interest transfer failed");
-        if(gdUBI > 0)
-            //transfer ubi to avatar on sidechain via bridge
-            require(token.transferAndCall(
-                bridgeContract,
-                gdUBI,
-                abi.encodePacked(ubiRecipient)
-            ),"ubi bridge transfer failed");
-        emit FundsTransferred(
-            msg.sender,
-            address(_staking),
-            reserveAddress,
-            interest,
-            gdUBI
-        );
-    }
-    /**
-     * @dev Mint to users reward tokens which they earned by staking contract
-     * @dev _user user to get rewards
-     */
-     function mintReward(
-        address _token,
-        address _user
-
-     ) public {
-        
-        Reward memory staking = rewardsForStakingContract[address(msg.sender)];
-        require(staking.blockStart > 0 , "Staking contract not registered");
-        uint amount = StakingContract(address(msg.sender)).rewardsMinted(_user);
-        if(amount > 0 && staking.isBlackListed == false){
-            
-            GoodReserveCDai(nameService.addresses(nameService.RESERVE())).mintRewardFromRR(_token, _user, amount);
-        }
-        
+			iToken.balanceOf(nameService.addresses(nameService.RESERVE())) -
+				currentBalance;
 
+		// Mints gd while the interest amount is equal to the transferred amount
+		uint256 gdUBI =
+			GoodReserveCDai(nameService.addresses(nameService.RESERVE()))
+				.mintUBI(
+				iToken,
+				interest // interest
+			);
+		// Transfers the minted tokens to the given staking contract
+		IGoodDollar token = IGoodDollar(address(avatar.nativeToken()));
 
-     }
-    
-    /**
-     * @dev Making the contract inactive after it has transferred funds to `_avatar`.
-     * Only the avatar can destroy the contract.
-     */
-  /**  function end() public onlyAvatar {
+		if (gdUBI > 0) {
+			//transfer ubi to avatar on sidechain via bridge
+			require(
+				token.transferAndCall(
+					bridgeContract,
+					gdUBI,
+					abi.encodePacked(ubiRecipient)
+				),
+				"ubi bridge transfer failed"
+			);
+		}
+		lastCollectedInterest = block.timestamp;
+		uint256 gasCostToMintReward = 200000; // Gas cost to mint GD reward to keeper hardcoded so should be changed according to calculations
```

Comment:
needs to be in a settable variable
find out what is the correct initial value

Reply:
I did and in the tests max amount was 130k so I set to 140k to be safe